src/AbstractPathMappingRepository.php

```diff
@@ -181,6 +213,31 @@ protected function createResource($filesystemPath, $path = null)
     }
 
     /**
+     * Create a link resource to another resource of the repository.
+     *
+     * @param string $targetPath The target path.
+     * @param string $path       The repository path.
+     *
+     * @return LinkResource The link resource.
+     *
+     * @throws RuntimeException If the targeted resource does not exist.
+     */
+    protected function createLinkResource($targetPath, $path = null)
+    {
+        if ($this->contains($targetPath)) {
+            $resource = new LinkResource($targetPath);
+            $resource->attachTo($this, $path);
+
+            return $resource;
+        }
+
+        throw new RuntimeException(sprintf(
+            'Trying to create a LinkResource on a non-existing target "%s"',
```

Comment:
I don't think we should throw an exception here. We'll get an exception once the link is resolved to a target, which seem sufficient to me.